lading/src/blackhole/datadog.rs

```diff
@@ -323,8 +324,14 @@ async fn handle_v2_protobuf(
                             let value = point.value.round() as u64;
                             counter_incr(&series.metric, &tag_pairs, value, timestamp).await
                         }
-                        2 | 3 => {
-                            // RATE | GAUGE
+                        2 => {
+                            // RATE
+                            #[allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+                            let val = (point.value * series.interval.max(1) as f64).round() as u64;
```

Comment:
```suggestion
let interval = series.interval;
if interval <= 0 {
    warn!("RATE with non-positive interval for {}", series.metric);
    continue;
}
let val = (point.value * interval.max(1) as f64).round() as u64;
```

something like this, the agent should NEVER emit a non-positive rate, lets fail louder.